### ISSUE TITLE
Padding functionality for numpy.roll()

### ISSUE THREAD

**Issue by Moe82** [numpy/numpy#17573]
I hope I'm posting in the right place. I'm looking to make my first contribution to an open source project and I believe I found a something that would improve the NumPy library. Wanted to get some opinions on it first before I dive into it. 

The numpy.roll() method rolls an array along a given axis without padding. It might be helpful if it accepted an argument that allows for padding.

For example:

numpy.roll(a, 1, axis=None, pad = 0)

Would shift the array 1 place to the right and change the first element to 0 instead of re-introducing the element that rolled beyond the last position at the first. One instance where this would helpful is image augmentation.

**Comment by eric-wieser** [comment]
This might be better suited to the mailing list, but isn't totally out of place here.

I'm not sure that `roll` is a great name for the operation you're describing - `roll` suggests a circular motion, but what you're doing here is discarding the circular bit!

I might be more keen on `np.pad(a, (1, -1), constant_values=0)` as a spelling for that operation - right now, `np.pad` does not accept negative indices, but that seems like a fairly natural extension.

**Comment by Moe82** [comment]
Makes sense- `roll()` suggests a circular roll and should do just that. I guess that's why it didn't have a padding functionality to begin with. 

Thats an interesting extension and makes sense to me. I read through the "Contributing to NumPy" article on numpy.org and will give it a shot.
